coco/check.py

```diff
@@ -377,23 +377,15 @@ async def run(self, result: Result):
             if r:
                 reply.update(r)
 
-        for host, result_ in reply.items():
-            if not result_:
-                if self.state_paths:
+        if self.state_paths:
+            for host, result_ in reply.items():
+                if not result_:
                     for name in self.state_paths.keys():
                         logger.debug(
                             f"/{self._name}: Missing value '{name}' in reply from {host}."
                         )
                         failed_hosts.add(host)
                         result.report_failure(self._name, host, "missing", name)
```

Comment:
there used to be a `continue` at the end of this block. Won't `result_.items()` fail `if not result_`?

Reply:
Thanks, good catch!